### DISCUSSION TITLE
Understanding Compliance Script Reset

### DISCUSSION THREAD

**Discussion by MbowerUD** [usnistgov/macos_security#404] [Q&A] [answered]
I just got hired on at my job and have been tasked with applying the second phase of CIS level 2, which we have reserved mostly for the things that require the compliance script to change.  I have properly issued this and got it to deploy, however I was informed that we are supposed to have some ability to revert the changes and that some of the things the compliance script changes we will have to use a script to change back.  I saw we had smart groups set up with --reset and --reset-all and was curious on what this actually does, is it just repopulating the compliance script or is it removing things that you set to exemptions that weren't exempted previously? As well as the best method for reverting changes made by the compliance script if --reset isn't the answer?

**Comment by jmahlman** [comment, marked as answer]
Hey there. 

Reset and reset all will reset/remove the audit files on the endpoints. Reset will remove the audit for the current baseline script, reset-all will remove all audit files regardless of script. 

There are no options to restore previous state as the code is not developed to do that. In order to restore previous state that would require some sort of method to know the previous state for each check (or some sort of method for restoring "default" state).
